### PR TITLE
add a simple Makefile

### PR DESCRIPTION
**Pull request by TypicalFence**
its a little silly

### CODE REVIEW

**Review by dd86k — CHANGES_REQUESTED** [review]
I believe it'd be better to `make debug` (default) and `make release`.

**Comment by TypicalFence** [comment]
added a debug target and a debugflags depending on the compiler used

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+.PHONY: clean install uninstall debug release
+
+default: debug
+
+debug:
+	$(MAKE) ddcpuid DFLAGS="$(DFLAGS) $(DEBUGFLAGS)"
+
+release:
+	$(MAKE) ddcpuid DFLAGS="$(DFLAGS)"
```
Comment (dd86k) [inline Makefile]: Looking good! Last change I'd like is appending `-O -release -boundscheck=off` to this line (at least checked on DMD and LDC, dunno about GDC).

**Review by dd86k — CHANGES_REQUESTED** [review]
Sorry again,

**Comment by TypicalFence** [comment]
It now looks like gdc won't compile tho

```
fence@fence-desktop ~/c/d/ddcpuid> make release DC=gdc
make ddcpuid DFLAGS=" -O -frelease -fbounds-check=off"
make[1]: Entering directory '/home/fence/code/d/ddcpuid'
gdc -O -frelease -fbounds-check=off ddcpuid.d
* sizeof CPUINFO: 224LU
* sizeof CACHE: 16LU
ddcpuid.d: Assembler messages:
ddcpuid.d:1014: Error: bad register name `%eaxmov %ebx'
make[1]: *** [Makefile:45: ddcpuid] Error 1
make[1]: Leaving directory '/home/fence/code/d/ddcpuid'
make: *** [Makefile:30: release] Error 2
```

dmd and ldc work fine.

if you wanna be fancy you could add some CI thing that builds it with all compilers :3

**Comment by dd86k** [comment]
Heh I haven't updated the inline GAS assembly in a while. It's a total mess, I think it was working before GDC 8.4.0. I'll have to fix that up eventually. 

The rest looks good.

**Review by dd86k — APPROVED** [review]

**Pull request merged**